ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -486,18 +486,15 @@ module Intel = struct
 			~is_host_display_enabled
 			~is_pci_hidden =
 		let types =
-			if is_system_display_device
-			then begin
-				match is_host_display_enabled, is_pci_hidden with
-				| false, true -> [passthrough_gpu]
-				| true, true -> []
-				| _, false ->
-					(make_vgpu_types ~__context
-						~pci ~whitelist:!Xapi_globs.gvt_g_whitelist)
-			end else
-				passthrough_gpu ::
-				(make_vgpu_types ~__context
-					~pci ~whitelist:!Xapi_globs.gvt_g_whitelist)
+			let passthrough_types =
+				match is_system_display_device, is_host_display_enabled, is_pci_hidden
+				with
+				| false, _ , _
+				| true, false, true -> [passthrough_gpu]
+				| _, _, _ -> []
+			in
+			passthrough_types @
+			(make_vgpu_types ~__context ~pci ~whitelist:!Xapi_globs.gvt_g_whitelist)
```

Comment:
The match statement means I have to keep track of what's what. Given they're all boolean values wouldn't it be jsut as clean to use if statements?

```ocaml
let passthrough_types = 
  if not is_system_display_device || not is_host_display_enabled && is_pci_hidden
  then [passthrough_gpu]
  else [] in
...
```
The match case with `false, _, _` and then `true, [other stuff]` are collapsed.